packages/cursorless-engine/src/core/commandRunner/CommandRunnerImpl.ts

```diff
@@ -78,6 +63,214 @@ export class CommandRunnerImpl implements CommandRunner {
+        const targets = this.getTargets(
+          inferenceContext,
+          partialActionDescriptor.target,
+          undefined,
+          this.actions.rewrapWithPairedDelimiter.getFinalStages(),
+        );
+        return this.actions.rewrapWithPairedDelimiter.run(
+          targets,
+          partialActionDescriptor.left,
+          partialActionDescriptor.right,
+        );
+      }
+      case "pasteFromClipboard": {
+        const destinations = this.getDestinations(
+          inferenceContext,
+          partialActionDescriptor.destination,
+        );
+        return this.actions.pasteFromClipboard.run(destinations);
+      }
+      case "executeCommand": {
+        const targets = this.getTargets(
+          inferenceContext,
+          partialActionDescriptor.target,
+        );
+        return this.actions.executeCommand.run(
+          targets,
+          partialActionDescriptor.commandId,
+          partialActionDescriptor.options,
+        );
+      }
+      case "replace": {
+        const targets = this.getTargets(
+          inferenceContext,
+          partialActionDescriptor.target,
+        );
+        return this.actions.replace.run(
+          targets,
+          partialActionDescriptor.replaceWith,
+        );
+      }
+      case "generateSnippet": {
+        const targets = this.getTargets(
+          inferenceContext,
+          partialActionDescriptor.target,
+        );
+        return this.actions.generateSnippet.run(
+          targets,
+          partialActionDescriptor.snippetName,
+        );
+      }
+      case "insertSnippet": {
+        const targets = this.getTargets(
+          inferenceContext,
+          partialActionDescriptor.target,
+          this.actions.insertSnippet.getPrePositionStages(
+            partialActionDescriptor.snippetDescription,
+          ),
+        );
+        return this.actions.insertSnippet.run(
+          targets,
+          partialActionDescriptor.snippetDescription,
+        );
+      }
+      case "wrapWithSnippet": {
+        const targets = this.getTargets(
+          inferenceContext,
+          partialActionDescriptor.target,
+          undefined,
+          this.actions.wrapWithSnippet.getFinalStages(
+            partialActionDescriptor.snippetDescription,
+          ),
+        );
+        return this.actions.wrapWithSnippet.run(
+          targets,
+          partialActionDescriptor.snippetDescription,
+        );
+      }
+      default: {
+        const action = this.actions[partialActionDescriptor.name];
+        const targets = this.getTargets(
+          inferenceContext,
+          partialActionDescriptor.target,
+          undefined,
+          action.getFinalStages?.(),
+        );
+        return action.run(targets);
+      }
+    }
+  }
+
+  private getTargets(
+    inferenceContext: InferenceContext,
+    partialTargetsDescriptor: PartialTargetDescriptor,
+    actionPrePositionStages?: ModifierStage[],
+    actionFinalStages?: ModifierStage[],
```

Comment:
I'm curious how this distinction works now that positions aren't modifiers, other than start of and end of

Reply:
Yeah we don't need preposition stages anymore